src/core/Table/filters/NumberRangeFilter/NumberRangeFilter.test.tsx

```diff
@@ -0,0 +1,143 @@
+    '.iui-input-container.iui-inline input',
+  ) as NodeListOf<HTMLInputElement>;
+  expect(labeledInputs.length).toBe(2);
+
+  expect(labeledInputs[0].value).toEqual('1');
+  expect(labeledInputs[1].value).toEqual('3');
+});
+
+it('should set filter when both values entered', () => {
+  const setFilter = jest.fn();
+  const { container } = renderComponent({ setFilter });
+
+  const labeledInputs = container.querySelectorAll(
+    '.iui-input-container.iui-inline input',
+  ) as NodeListOf<HTMLInputElement>;
+  expect(labeledInputs.length).toBe(2);
+
+  fireEvent.change(labeledInputs[0], { target: { value: '1' } });
+  fireEvent.change(labeledInputs[1], { target: { value: '3' } });
+
+  screen.getByText('Filter').click();
+
+  expect(setFilter).toHaveBeenCalledWith([1, 3]);
+});
+
+it('should set filter when only From is entered', () => {
+  const setFilter = jest.fn();
+  const { container } = renderComponent({ setFilter });
+
+  const labeledInputs = container.querySelectorAll(
+    '.iui-input-container.iui-inline input',
+  ) as NodeListOf<HTMLInputElement>;
+  expect(labeledInputs.length).toBe(2);
+
+  fireEvent.change(labeledInputs[0], { target: { value: '1' } });
+
+  screen.getByText('Filter').click();
+
+  expect(setFilter).toHaveBeenCalledWith([1, undefined]);
+});
+
+it('should set filter when only To is entered', () => {
+  const setFilter = jest.fn();
+  const { container } = renderComponent({ setFilter });
+
+  const labeledInputs = container.querySelectorAll(
+    '.iui-input-container.iui-inline input',
+  ) as NodeListOf<HTMLInputElement>;
+  expect(labeledInputs.length).toBe(2);
+
+  fireEvent.change(labeledInputs[1], { target: { value: '3' } });
+
+  screen.getByText('Filter').click();
+
+  expect(setFilter).toHaveBeenCalledWith([undefined, 3]);
+});
+
+it('should set filter when both values entered and Enter is pressed', () => {
+  const setFilter = jest.fn();
+  const { container } = renderComponent({ setFilter });
+
+  const labeledInputs = container.querySelectorAll(
+    '.iui-input-container.iui-inline input',
+  ) as NodeListOf<HTMLInputElement>;
+  expect(labeledInputs.length).toBe(2);
+
+  fireEvent.change(labeledInputs[0], { target: { value: '1' } });
+  fireEvent.change(labeledInputs[1], { target: { value: '3' } });
+
+  fireEvent.keyDown(labeledInputs[1], {
+    key: 'Enter',
+    charCode: 13,
+  });
+
+  expect(setFilter).toHaveBeenCalledWith([1, 3]);
+});
+
+it('should set filter with empty values when invalid date is entered', () => {
+  const setFilter = jest.fn();
+  const { container } = renderComponent({ setFilter });
+
+  const labeledInputs = container.querySelectorAll(
+    '.iui-input-container.iui-inline input',
+  ) as NodeListOf<HTMLInputElement>;
+  expect(labeledInputs.length).toBe(2);
+
+  fireEvent.change(labeledInputs[0], { target: { value: 'abc' } });
+
+  screen.getByText('Filter').click();
+
+  expect(setFilter).toHaveBeenCalledWith([undefined, undefined]);
```

Comment:
I dont think we should allow to call set with wrong values..

Reply:
With `type='number'` it doesn't allow you to enter anything except number. It is more just to be sure if some browsers (I guess IE) don't handle number inputs.